Expand on Conda CUDA packaging history

jakirkham opened the pull request:
Read through this section. There was already some good stuff here.

Tried to provide some more background on earlier CUDA packaging work. Also tried to contextualize changes in the packaging structure and what they attempted to resolve.

Am sure there is more we can revise/add here. Though this seemed like a good point to pause and hear some feedback.

matthewfeickert commented:
> Also happy to apply line wrapping. Was trying not to make the diff too crazy. Though long lines can be hard to comment on. So please let me know what you think

:sweat_smile: Yeah, no worries. I go for "1 sentence per line" because I have academic-LaTeX-brain, but I fully appreciate that "text per line" is as personal / decisive a subject as text editors. :)

jakirkham commented:
Thanks Matthew! 🙏

Have incorporated your feedback. Also read back through and fixed some wording

Added a couple more lines, but the bulk of the content is the same

Please let me know if you have any thoughts on this
